examples/tree_source/tree_source/app.py

```diff
@@ -1,119 +1,150 @@
-from random import choice
+import os
 
 import toga
 from toga.style import Pack
-from toga.constants import ROW, COLUMN
+from toga.constants import COLUMN
 from toga.sources import Source
+from datetime import datetime
 
-bee_movies = [
-    {'year': 2008, 'title': 'The Secret Life of Bees', 'rating': '7.3', 'genre': 'Drama'},
-    {'year': 2007, 'title': 'Bee Movie', 'rating': '6.1', 'genre': 'Animation, Adventure, Comedy'},
-    {'year': 1998, 'title': 'Bees', 'rating': '6.3', 'genre': 'Horror'},
-    {'year': 2007, 'title': 'The Girl Who Swallowed Bees', 'rating': '7.5', 'genre': 'Short'},
-    {'year': 1974, 'title': 'Birds Do It, Bees Do It', 'rating': '7.3', 'genre': 'Documentary'},
-    {'year': 1998, 'title': 'Bees: A Life for the Queen', 'rating': '8.0', 'genre': 'TV Movie'},
-    {'year': 1994, 'title': 'Bees in Paradise', 'rating': '5.4', 'genre': 'Comedy, Musical'},
-    {'year': 1947, 'title': 'Keeper of the Bees', 'rating': '6.3', 'genre': 'Drama'}
-]
-
-class Movie:
-    # A class to wrap individual
-    def __init__(self, year, title, rating, genre):
-        self.year = year
-        self.title = title
-        self.rating = rating
-        self.genre = genre
-
-
-class Decade:
-    # A class to wrap
-    def __init__(self, decade):
-        self.decade = decade
-        self._data = []
-
-    # Display values for the decade in the tree.
-    @property
-    def year(self):
-        return "{}0's".format(self.decade)
+
+class LoadingFailedNode:
+    """A node to represent failed loading of children"""
+
+    def __init__(self, parent):
+        self.parent = parent
+        self.children = []
+        self.name = 'loading failed'
+        self.date_modified = ''
 
     # Methods required for the data source interface
     def __len__(self):
-        return len(self._data)
+        return 0
 
     def __getitem__(self, index):
-        return self._data[index]
+        pass
+
+    @staticmethod
+    def can_have_children():
+        return False
 
-    def can_have_children(self):
-        return True
 
+class Node:
+    """A node which loads its children on-demand."""
 
-class DecadeSource(Source):
-    def __init__(self):
+    def __init__(self, path, parent):
         super().__init__()
-        self._decades = []
 
+        self.parent = parent
+        self._children = []
+
+        self.path = path
+        self._mtime = os.stat(self.path).st_mtime
+        self._icon = toga.Icon('resources/folder.icns')
```

Comment:
The `.icns` extension should be ommitted, to allow the end platform to find the best image. Adding a 32px and 72 px png, and a 32px ICO would be advisable.

Reply:
Will do. Does toga expect images with different resolutions to be saved in a particular way?